src/mappings/prices.ts

```diff
@@ -0,0 +1,59 @@
+import { log, BigInt } from '@graphprotocol/graph-ts'
+import { Order, Price, Trade } from '../../generated/schema'
+import { EthereumEvent } from '@graphprotocol/graph-ts'
+import { toPriceId } from '../utils'
+import { BatchExchange } from '../../generated/BatchExchange/BatchExchange'
+
+export function createOrUpdatePrice(tokenId: i32, trade: Trade, event: EthereumEvent): void {
+  log.info('[createOrUpdatePrice] Create or Update Price for batch {} and Token {}. Tx: ', [
+    trade.tradeBatchId.toString(),
+    BigInt.fromI32(tokenId).toString(),
+    event.transaction.hash.toHex(),
+  ])
+
+  let priceId = toPriceId(tokenId, trade.tradeBatchId)
+  let price = Price.load(priceId)
+
+  log.info('[createOrUpdatePrice] Make sure price {} exists', [priceId])
+  if (price == null) {
+    // Create price for the current batch
+    price = _createPrice(priceId, tokenId, trade, event)
+  }
+  // TODO: Else handling reverts of solutions
```

Comment:
Besides reverts, it seems that the price is never updated, only the volume. Is that also TODO?

Reply:
Oh I see. There's a price entry for every batch. So we'll only need to update the volume in the same batch, never the price.

I'll leave this comment for future reference.

Reply:
Yes, I'll make also an issue so we don't forget :)